build-test.sh

```diff
@@ -3,83 +3,171 @@
 ## Exit immediately on error.
 set -e
 
-# Echo to console.
-set -x
+# Set function to be called on exit
+trap 'exit_err_cleanup $?' EXIT
 
+###############################################################################
+# exit_err_cleanup is called at exit, if there was an error it calls container
+# cleanup and reports an error.
+# Globals: none
+# Parameters:
+# - $1: Script exit code
+###############################################################################
+exit_err_cleanup() {
+  # Cleanup containers and report an error if any happened
+  if [ "$1" != "0" ]; then
+    cleanup_test_containers
+    echo "ERROR: Error $1 occured"
+  else
+    echo "SUCCESS: Build and test finished successfully"
+  fi
+
+  exit $1
+}
+
+###############################################################################
+# cleanup_test_containers removes test containers.
+# Globals: none
+# Parameters: none
+###############################################################################
 cleanup_test_containers() {
+  doing "Removing test containers"
   CONTAINERS=$(docker ps -a -q --filter "name=sia-ant-farm-test-container")
   if [ ! -z $CONTAINERS ]; then
     docker stop $CONTAINERS
     docker rm   $CONTAINERS
   fi
+  finished
 }
 
 ###############################################################################
-# Echo curl loop string to be used in 'timeout' command. The string when
-# executed executes curl command with the given url in the loop untill the curl
-# executes successfully.
+# doing sets and echoes the given message.
 # Globals:
-#   none
+# - DOING_MSG: variable to store the message for finished() function
+# Parameters:
+# - $1: The message to be set and to be echoed
+###############################################################################
+doing() {
+  DOING_MSG=$1
+  echo "$DOING_MSG..."
+}
+
+###############################################################################
+# finished echoes that the message finished.
+# Globals:
+# - DOING_MSG: variable to store the message from doing() function
+# Parameters: none
+###############################################################################
+finished() {
+  echo "$DOING_MSG finished successfully"
```

Comment:
1. Should `finished` clear the message?
2. Can we avoid using globals?

Reply:
Resolved in the next PR:
Created a wrapper function to log start, call the command and log end. No more need to use globals.